src/plane.rs

```diff
@@ -188,52 +129,26 @@ impl<T: Pixel> PlaneData<T> {
     cfg_if::cfg_if! {
       if #[cfg(target_arch = "wasm32")] {
         // FIXME: wasm32 allocator fails for alignment larger than 3
-        const DATA_ALIGNMENT_LOG2: usize = 3;
+        const DATA_ALIGNMENT: usize = 1 << 3;
       } else {
-        const DATA_ALIGNMENT_LOG2: usize = 6;
+        const DATA_ALIGNMENT: usize = 1 << 6;
       }
     }
 
-    unsafe fn layout(len: usize) -> Layout {
-        Layout::from_size_align(len * mem::size_of::<T>(), 1 << Self::DATA_ALIGNMENT_LOG2)
-            .expect("layout size too large")
-    }
-
-    unsafe fn new_uninitialized(len: usize) -> Self {
-        let ptr = {
-            let layout = Self::layout(len);
-            let ptr = alloc(layout) as *mut T;
-            if ptr.is_null() {
-                handle_alloc_error(layout);
-            }
-            std::ptr::NonNull::new_unchecked(ptr)
-        };
-
-        PlaneData {
-            ptr,
-            len,
-            _marker: PhantomData,
-        }
-    }
-
     pub fn new(len: usize) -> Self {
-        // SAFETY: we initialize the plane data before returning
-        let mut pd = unsafe { Self::new_uninitialized(len) };
-
-        for v in pd.iter_mut() {
-            *v = T::cast_from(128);
+        Self {
+            data: AVec::from_iter(
+                Self::DATA_ALIGNMENT,
```

Comment:
After skimming through aligned-vec, it's fine (and maybe preferable?) to just pass 0 here. That way there'd be no need for another target-dependent const variable.